### PR TITLE
3 bugfixes

### LINKED ISSUE #238: Layer 0 editing becomes buggy again when it has a different size comparing to Layer 1 in beta-10

### LINKED ISSUE #239: Select a Door will change the Door's position
happens when mouse release, door's X1 and Y1 are set to mouse position

### PR DESCRIPTION
**Pull request by shinespeciall**
Fixed #238 , Fixed #239 .
Fixed cannot avoid creating new room when the current room is a new room which has not been saed into the rom, causing read overflow.

### CODE REVIEW

File: WL4EditorWindow.cpp [hunk 1]
```diff
@@ -1917,6 +1917,16 @@ void WL4EditorWindow::on_actionNew_Room_triggered()
         OutputWidget->PrintString("Cannot add more Room to the current Level!");
         return;
     }
+    int offset = WL4Constants::LevelHeaderIndexTable + selectedLevel._PassageIndex * 24 + selectedLevel._LevelIndex * 4;
+    int levelHeaderIndex = ROMUtils::IntFromData(offset);
+    int levelHeaderPointer = WL4Constants::LevelHeaderTable + levelHeaderIndex * 12;
+    int roomCount = ROMUtils::CurrentFile[levelHeaderPointer + 1];
+    if (roomCount <= static_cast<int>(selectedRoom))
+    {
+        OutputWidget->PrintString("Cannot create room, current Room has not been saved in th rom yet!");
```
Comment (Goldensunboy) [inline WL4EditorWindow.cpp:1926]: the

File: LevelComponents/Room.cpp [hunk 2]
```diff
@@ -849,18 +854,26 @@ namespace LevelComponents
                         QImage imageA = RenderedLayers[0]->pixmap().toImage();
                         QImage imageB = alphaPixmapTemp.toImage();
                         for(auto iter: renderParams->tilechangelist) {
+                            int substitudeEVA = Layer0ColorBlendCoefficient_EVA;
```
Comment (Goldensunboy) [inline LevelComponents/Room.cpp:857]: substituteEVA

File: WL4EditorWindow.cpp (hunk 1 above)
Comment (shinespeciall) [reply WL4EditorWindow.cpp]: Done.

File: LevelComponents/Room.cpp (hunk 2 above)
Comment (shinespeciall) [reply LevelComponents/Room.cpp]: Done.

**Pull request merged**